lib/cli/watch-run.js

```diff
@@ -136,6 +137,43 @@ exports.watchRun = (mocha, {watchFiles, watchIgnore}, fileCollectParams) => {
   });
 };
 
+class GlobFilesTracker {
+  constructor(watchFiles, watchIgnore) {
+    this.watchFilesSet = new Set();
+    this.watchFiles = watchFiles;
+    this.watchIgnore = watchIgnore;
+  }
+
+  regenerate() {
+    const watchIgnoreSet = new Set();
+    for (const pattern of this.watchIgnore) {
+      glob.sync(pattern, { dot: true }).forEach(filePath => watchIgnoreSet.add(filePath));
+    }
+
+    const globOpts = {
+      dot: true,
+      ignore: {
+        ignored: () => false,
```

Comment:
Despite the docs suggesting the object needs both `childrenIgnored` and `ignored`, I omitted `ignored` with no adverse effects locally. I think we can remove this?

```suggestion
```